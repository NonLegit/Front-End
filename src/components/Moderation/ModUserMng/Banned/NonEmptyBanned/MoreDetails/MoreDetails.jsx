import { Box, Typography } from '@mui/material';
import { StyledDetails } from '../../../styles';

/**
 * more datails tab
 * @component
 * @property  {boolean} isOpened determine the state of more details tab to be showed or hidden
 * @property {enum} bannedFor explain why this user is banned (spam or threatining etc...)
 * @property {string} modNote note from the moderator about the banned user
 * @return {React.Component} - more details tab
 */

function MoreDetails(props) {
  const { modNote, bannedFor, isOpened } = props;
  return (
    <StyledDetails
      display={(isOpened) ? 'flex' : 'none'}
    >
      <Box padding="16px">
        {(modNote !== '') && (
          <Box display="flex" alignItems="center" paddingBottom="10px">
            <Typography
              fontSize="10px"
              color="#878A8C"
              fontWeight={700}
              marginRight="3px"
              textTransform="uppercase"
            >
              mod note:
            </Typography>
            <Typography
              fontSize="12px"
              color="#1C1C1C"
              paddingBottom="2px"
            >
              {modNote}
            </Typography>
          </Box>
        )}
        <Box display="flex" alignItems="center">
          <Typography
            fontSize="10px"
            color="#878a8c"
            fontWeight={700}
            marginRight="3px"
            textTransform="uppercase"
          >
            banned for:
          </Typography>
          <Typography
            fontSize="12px"
            color="#1C1C1C"
            paddingBottom="2px"
          >
            {bannedFor}
          </Typography>
        </Box>
      </Box>
    </StyledDetails>
  );
}

export default MoreDetails;
