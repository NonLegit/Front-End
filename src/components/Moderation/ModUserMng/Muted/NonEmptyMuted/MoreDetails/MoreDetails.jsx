import { Box, Typography } from '@mui/material';
import { StyledDetails } from '../../../styles';

/**
 * more datails tab
 * @component
 * @property  {boolean} isOpened determine the state of more details tab to be showed or hidden
 * @property {string} modNote note from the moderator about the muted user
 * @return {React.Component} - more details tab
 */

function MoreDetails(props) {
  const { modNote, isOpened } = props;
  return (
    <StyledDetails
      display={(isOpened) ? 'flex' : 'none'}
    >
      <Box padding="16px" display="flex" alignItems="center" paddingBottom="10px">
        {(() => {
          if (modNote === '') {
            return (
              <Typography
                fontSize="10px"
                color="#878A8C"
                fontWeight={700}
                marginRight="3px"
                textTransform="uppercase"
              >
                No mod Note.
              </Typography>
            );
          }
          return (
            <>
              <Typography
                fontSize="10px"
                color="#878A8C"
                fontWeight={700}
                marginRight="3px"
                textTransform="uppercase"
              >
                mod note:
              </Typography>
              <Typography
                fontSize="12px"
                color="#1C1C1C"
                paddingBottom="2px"
              >
                {modNote}
              </Typography>
            </>
          );
        })()}

      </Box>
    </StyledDetails>
  );
}

export default MoreDetails;
