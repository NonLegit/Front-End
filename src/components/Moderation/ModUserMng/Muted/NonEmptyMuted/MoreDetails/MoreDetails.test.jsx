import renderer from 'react-test-renderer';
import MoreDetails from './MoreDetails';

test('test snapshot', async () => {
  const tree = renderer.create(<MoreDetails />).toJSON();
  expect(tree).toMatchSnapshot();
});
