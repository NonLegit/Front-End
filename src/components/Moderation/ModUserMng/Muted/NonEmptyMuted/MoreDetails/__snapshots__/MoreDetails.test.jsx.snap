// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-21nkcj"
>
  <div
    className="MuiBox-root css-1rzk8ps"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-16acbxi-MuiTypography-root"
    >
      mod note:
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1gaz5e1-MuiTypography-root"
    />
  </div>
</div>
`;
